### PR TITLE
corehost: Build intermediate library to avoid building twice

### PR DESCRIPTION
**Pull request by lpereira**
corehost: Build intermediate libraries to avoid building twice

Some files are built twice in the hosting layer, so create two separate static libraries that are then linked with all targets that need them.

Fixes [core-setup/8276](https://github.com/dotnet/core-setup/issues/8276).

### CODE REVIEW

**Comment by lpereira** [comment]
(Please refrain from reviewing until I fix the build issues on Windows et al)

**Comment by lpereira** [comment]
The Linux_arm failure seems unrelated (unable to download stuff needed to build), but the build is now passing on Windows, too.  @jkoritzinsky, I'd appreciate your input here.

File: src/installer/corehost/cli/common/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Licensed to the .NET Foundation under one or more agreements.
+# The .NET Foundation licenses this file to you under the MIT license.
+# See the LICENSE file in the project root for more information.
+
+project(common)
```
Comment (jkoritzinsky) [inline src/installer/corehost/cli/common/CMakeLists.txt]: Can we rename common here? It seems like it's basically just the hosting PAL and long-file support.

**Review by jkoritzinsky — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Yep the Linux arm failure is unrelated: #1880

File: src/installer/corehost/cli/common/CMakeLists.txt (hunk 1 above)
Comment (lpereira) [reply src/installer/corehost/cli/common/CMakeLists.txt]: Will do.

Comment (lpereira) [reply src/installer/corehost/cli/common/CMakeLists.txt]: Done; called it `hostmisc` for the lack of a better name.  Let's see what CI has to say this time.

**Comment by jkoritzinsky** [comment]
Runtime-live-build failure is a known issue. I'll merge this in.

**Pull request merged**